codegens/http/lib/code-http-converter.js

```diff
@@ -29,18 +30,33 @@ function getOptions () {
  */
 function convert (request, options, callback) {
   let snippet = '',
-    url = request.url.toString(),
-    body;
+    url, host, path, query, body;
   options = utils.sanitizeOptions(options, getOptions());
-  // urlParser expects a protocal in the url
-  // If it is not present we have to add it manually
-  if (!url.match(/^([a-z][a-z0-9.+-]*:)?(\/\/)/)) {
-    url = `http://${url}`;
+
+  try {
+    url = sdk.Url.parse(request.url);
+  }
+  catch (e) {
+    url = request.url;
```

Comment:
@webholik Can we have a unit test testing this case?

Reply:
I added this when I was testing `URL` class in Nodejs, which throws an exception if url is malformed. But I checked `postman-collection` source and `Url.parse` doesn't throw any exception so I have removed the try catch here.